### ISSUE TITLE
read of @NonNull field before initialization doesn't recognize earlier assignment in the constructor

### ISSUE THREAD

**Issue by ZacSweers** [uber/NullAway#287]
Consider this case

```java
private final SizeUpdateListener sizeUpdateListener;

DiskKeyValueStore(
    SizeUpdateListener sizeUpdateListener{
  this.sizeUpdateListener = sizeUpdateListener;
  this.sizeUpdateListener.onSizeUpdateListenerConnect();
}
```

NullAway will report an error for the initial assignment however, and instead reports this:

```
Command failed with exit code 1.
stderr: .../DiskKeyValueStore.java:86: warning: [NullAway] read of @NonNull field sizeUpdateListener before initialization
    this.sizeUpdateListener.onSizeUpdateListenerConnect();
        ^
    (see http://t.uber.com/nullaway )
: error: warnings found and -Werror specified

Errors: 1. Warnings: 1.
```

**Comment by msridhar** [comment]
This is definitely a bug. We will take a look.

**Comment by msridhar** [comment]
I tried to repro with a test in f1227cae536b3e39801d2bf87ea3a32093d9f8bd but I didn't see any warning.  @ZacSweers any further details on how to repro?

**Comment by ZacSweers** [comment]
So this seems to only happen when I try to run tests using this, and not
when I just build it myself. Unfortunately the repro case is internal code
but I've detailed it to Lazaro on the internal diff

**Comment by msridhar** [comment]
Ok assigning to @lazaroclapp

**Comment by ZacSweers** [comment]
Good news - this is fake news, thanks to weird buck daemon issues. Lazaro and I were able to repro, then nuking my local cache auto-magically fixed it. Even better, it would only repro if you added a `--filter` flag during the test build.

🎶 Just buck things 🎶

**Comment by idanakav** [comment]
3 years later, it's my turn to face the issue after we recently removed the suppression.
In my case, I was seeing it during regular builds, and nuking the local cache helped once but then it came back.

**Comment by msridhar** [comment]
@idanakav if you think this is a NullAway bug, it would be great if you could provide an open-source test case to reproduce the problem.  If you think it's an issue with the build config at Uber, ping @lazaroclapp and he can hopefully help you out
